### ISSUE TITLE
When logging in with orcid, you get an error page (logs in ok though)

### ISSUE THREAD

**Issue by david-caro** [HEPData/hepdata#106]
You get the following dump on the logs:
```python
Traceback (most recent call last):
  File "/opt/hepdata/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/hepdata/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/hepdata/lib/python2.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/hepdata/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/hepdata/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/hepdata/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/opt/hepdata/lib/python2.7/site-packages/invenio_oauthclient/views/client.py", line 129, in authorized
    return current_oauthclient.handlers[remote_app]()
  File "/opt/hepdata/lib/python2.7/site-packages/flask_oauthlib/client.py", line 699, in decorated
    return f(*((data,) + args), **kwargs)
  File "/opt/hepdata/lib/python2.7/site-packages/invenio_oauthclient/handlers.py", line 501, in inner
    return f(args[0], remote, *args[1:], **kwargs)
  File "/opt/hepdata/lib/python2.7/site-packages/invenio_oauthclient/handlers.py", line 234, in inner
    return f(*args, **kwargs)
  File "/opt/hepdata/lib/python2.7/site-packages/invenio_oauthclient/handlers.py", line 330, in authorized_signup_handler
    [remote.name].get('remember', False)):
  File "/opt/hepdata/lib/python2.7/site-packages/invenio_oauthclient/utils.py", line 101, in oauth_authenticate
    if login_user(user):
  File "/opt/hepdata/lib/python2.7/site-packages/flask_security/utils.py", line 88, in login_user
    identity=Identity(user.id))
  File "/opt/hepdata/lib/python2.7/site-packages/blinker/base.py", line 267, in send
    for receiver in self.receivers_for(sender)]
  File "/opt/hepdata/lib/python2.7/site-packages/invenio_oauthclient/contrib/cern.py", line 351, in on_identity_changed
    account_groups(account, resource, refresh_timedelta=refresh)
  File "/opt/hepdata/lib/python2.7/site-packages/invenio_oauthclient/contrib/cern.py", line 219, in account_groups
    groups = fetch_groups(resource['Group'])
KeyError: 'Group'
```

**Comment by GraemeWatt** [comment]
Similar report from @clelange for a HEPData account with a CERN e-group email address linked to both CERN and ORCID accounts:

> There are some issues with the HEPData website:
> - when I click on "Security" on the account dashboard, I get the error listed below
> - the same thing happens when I log in (but I'm logged in alright)
> - it also happens when changing the password (but changing it worked)
>
> Error:
> Sorry, we have some issues processing your request.
> If you have any questions, please email info@hepdata.net.
>
> KeyError: 'Group'

[Disconnecting](https://www.hepdata.net/account/settings/linkedaccounts/) the CERN and ORCID logins gives #146 when clicking on "[Security](https://www.hepdata.net/account/settings/security/)".  

Try upgrading the Invenio packages (#138), particularly [invenio-accounts](https://github.com/inveniosoftware/invenio-accounts).

**Comment by GraemeWatt** [comment]
This might be fixed after upgrading [Invenio-OAuthClient](https://github.com/inveniosoftware/invenio-oauthclient) and switching to the new CERN SSO (see #297).

**Comment by alisonrclarke** [comment]
This also seems to happen if you log in with username and password when you have a linked CERN account, if you aren't already logged in to CERN.

I think this is actually an issue with invenio_oauthclient, as it seems to call the post login handlers for all linked accounts even if you've logged in by a different means, and the CERN handlers make further API calls to get user details. It looks like it's already been logged as inveniosoftware/invenio-oauthclient/issues/228

Unfortunately a similar issue occurs locally even with the latest invenio-oathclient and the new CERN SSO (after working on #297).

**Comment by GraemeWatt** [comment]
I've checked again and there's still an issue with the new CERN SSO and the latest `invenio-oauthclient` v1.5.0.  Logging in with email address and local password to an account that's also linked to a CERN account gives `KeyError: 'cern_roles'` from [this line](https://github.com/inveniosoftware/invenio-oauthclient/blob/2f4a4e60f595ab90fd60571dc49cc2d640cf806b/invenio_oauthclient/contrib/cern_openid.py#L194) of [cern_openid.py](https://github.com/inveniosoftware/invenio-oauthclient/blob/2f4a4e60f595ab90fd60571dc49cc2d640cf806b/invenio_oauthclient/contrib/cern_openid.py) in the `invenio-oauthclient` v1.5.0 package.  Yesterday I think I must have previously logged in with my CERN account and so I did not see the error.  I'll add a comment to inveniosoftware/invenio-oauthclient#228.
